src/asymmetric_crypto/curve_25519/x25519/encoding.rs

```diff
@@ -0,0 +1,60 @@
+use pkcs8::{
+    der::asn1::BitStringRef, spki, Document, EncodePrivateKey, EncodePublicKey, ObjectIdentifier,
+    PrivateKeyInfo, SecretDocument,
+};
+
+use super::key_pair::X25519Keypair;
+use crate::{RandomFixedSizeCBytes, X25519PublicKey, CURVE_25519_PRIVATE_KEY_LENGTH};
+
+impl EncodePrivateKey for X25519Keypair {
+    fn to_pkcs8_der(&self) -> pkcs8::Result<pkcs8::SecretDocument> {
+        // Serialize private key as nested OCTET STRING
+        let mut private_key = [0u8; 2 + CURVE_25519_PRIVATE_KEY_LENGTH];
+        private_key[0] = 0x04;
+        private_key[1] = 0x20;
+        private_key[2..].copy_from_slice(self.private_key.as_bytes());
+
+        let private_key_info = PrivateKeyInfo {
+            algorithm: X25519_ALGORITHM_ID,
+            private_key: &private_key,
+            public_key: None,
+        };
```

Comment:
In order to get the zeroization, you should write something like that:
```suggestion
        let mut private_key = Zeroizing::new([0u8; 2 + CURVE_25519_PRIVATE_KEY_LENGTH]);
        private_key[0] = 0x04;
        private_key[1] = 0x20;
        private_key[2..].copy_from_slice(self.private_key.as_bytes());

        let private_key_info = PrivateKeyInfo {
            algorithm: X25519_ALGORITHM_ID,
            private_key: private_key.as_slice(),
            public_key: None,
        };
```